offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/OfferToPDFConverter.groovy

```diff
@@ -201,6 +247,42 @@ class OfferToPDFConverter implements OfferExporter {
         htmlContent.getElementById("offer-date").text(dateFormat.format(offer.modificationDate))
     }
 
+    void setIntermediatePrices(boolean isOverhead) {
+
+        if (isOverhead) {
+            String overheadStatus = "overhead"
+
+            double itemsWithOverheadNetPrice = offer.getItemsWithOverheadNetPrice()
+            double overheadPrice = offer.getOverheads()
+            double totalPrice = itemsWithOverheadNetPrice + overheadPrice
+
+            final String formattedTotalPrice = Currency.getFormatterWithoutSymbol().format(totalPrice)
+            final String formattedItemsWithOverheadNetPrice = Currency.getFormatterWithoutSymbol().format(itemsWithOverheadNetPrice)
+            final String formattedOverheadPrice = Currency.getFormatterWithoutSymbol().format(overheadPrice)
+
+            htmlContent.getElementById("${overheadStatus}-value-total").text(formattedTotalPrice)
+            htmlContent.getElementById("${overheadStatus}-value-net").text(formattedItemsWithOverheadNetPrice)
+            htmlContent.getElementById("${overheadStatus}-value-overhead").text(formattedOverheadPrice)
+        } else {
+            String overheadStatus = "no-overhead"
+
+            //No Overhead Prices are calculated for these items
+            double overheadPrice = 0.00
+            double itemsWithoutOverheadNetPrice = offer.getItemsWithOverheadNetPrice()
+            double totalPrice = itemsWithoutOverheadNetPrice
+
+            final String formattedTotalPrice = Currency.getFormatterWithoutSymbol().format(totalPrice)
+            final String formattedItemsWithOverheadNetPrice = Currency.getFormatterWithoutSymbol().format(itemsWithoutOverheadNetPrice)
+            final String formattedOverheadPrice = Currency.getFormatterWithoutSymbol().format(overheadPrice)
+
+            htmlContent.getElementById("${overheadStatus}-value-total").text(formattedTotalPrice)
+            htmlContent.getElementById("${overheadStatus}-value-net").text(formattedItemsWithOverheadNetPrice)
+            htmlContent.getElementById("${overheadStatus}-value-overhead").text(formattedOverheadPrice)
+        }
+
+    }
+
+
```

Comment:
why not splitting this method in two?
one method for the if case and another method for the else case and then you can call them also more explicitly without the boolean as method argument

Reply:
Good Idea! 👍